### PR TITLE
Expose networkInfo, handle account and network changes

### LINKED ISSUE #27: web3: should provide enumerable network object
The web3 provider should give more information like `chainId`. It should also expose the name not only as a string but as an enum.

https://github.com/rsksmart/rif-ui/blob/master/src/providers/Web3Provider.tsx#L39

### LINKED ISSUE #28: web3: detect account changed
The page no longer reloads in many wallet when switching between accounts in the same wallet (e.g. metamask, nifty). This should be detected and the Web3Provider should appropriately provide new information.

### LINKED ISSUE #73: web3: detect networkChanged
At the moment, the browser reloads the page when the network changes. But MetaMask will soon stop doing that. We should detect this and the Web3Provider should act upon that.

### Console warning when switching networks
`MetaMask will soon stop reloading pages on network change.
If you rely upon this behavior, add a 'networkChanged' event handler to trigger the reload manually: https://metamask.github.io/metamask-docs/API_Reference/Ethereum_Provider#ethereum.on(eventname%2C-callback)`

### PR DESCRIPTION
**Pull request by farinalvaro**
This PR closes #27, closes #28 and closes #73 
- Created `NetworkInfo` interface and exposed the object through the `Web3Provider`.
- Handled `onAccountsChange`
- Handled `onNetworkChange`
- Added `NetworkIndicator` component
- Updated version number to `0.1.0-dev.4`

### CODE REVIEW

File: src/providers/Web3Provider.tsx [hunk 1]
```diff
@@ -83,22 +89,29 @@ class Web3Provider extends Component<{}, Web3ProviderState> {
     if (Array.isArray(accounts)) [account] = accounts
     else account = accounts
     let networkId = await web3.eth.net.getId()
-
-    if (networkId === 1) networkId = await web3.eth.getChainId()
+    let chainId: Number | undefined = undefined
+    if (networkId) chainId = await web3.eth.getChainId()
```
Comment (vojtechsimetka) [inline src/providers/Web3Provider.tsx:93]: This will throw on RSK.

**Comment by julianmrodri** [comment]
OK from a functional standpoint! Good job!

File: src/components/atoms/NetworkIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+const getNetworkStatus = (currentNetworkId, requiredNetworkId): NetworkStatus => {
+  if (currentNetworkId) {
+    if (currentNetworkId === requiredNetworkId) {
+      return NetworkStatus.RIGHT_NETWORK
+    }
+    return NetworkStatus.NETWORK_MISSMATCH
+  }
+  return NetworkStatus.NO_NETWORK
+}
```
Comment (vojtechsimetka) [inline src/components/atoms/NetworkIndicator.tsx:35]: I don't like this nesting
```suggestion
const getNetworkStatus = (currentNetworkId, requiredNetworkId): NetworkStatus => {
  if (!currentNetworkId) return NetworkStatus.NO_NETWORK
  if (currentNetworkId === requiredNetworkId) return NetworkStatus.RIGHT_NETWORK
  return NetworkStatus.NETWORK_MISSMATCH
}
```

File: src/providers/Web3Provider.tsx [hunk 3]
```diff
@@ -1,106 +1,196 @@
...
-      return 'xDai'
-    default:
-      return null
+const getAccountFromEthAccounts = (accounts: string[] | string): string | undefined => {
+  let account: string
+
+  if (Array.isArray(accounts)) [account] = accounts
+  else account = accounts
+  return account
+}
```
Comment (vojtechsimetka) [inline src/providers/Web3Provider.tsx:59]: Awesome!

File: src/components/atoms/NetworkIndicator.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+const defaultOnRightNetworkMessage = 'You are on the right network'
+const deaulftOnWrongNetworkMessage = 'You are on the wrong network'
+const defaultNoNetworkMessage = 'You are not connected to any network'
+
+export enum NetworkStatus {
+  NO_NETWORK = 1,
+  NETWORK_MISSMATCH = 2,
+  RIGHT_NETWORK = 3
```
Comment (vojtechsimetka) [inline src/components/atoms/NetworkIndicator.tsx:24]: Can we rename this to CORRECT_NETWORK ?

File: src/components/atoms/NetworkIndicator.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import CheckCircleOutlineOutlinedIcon from '@material-ui/icons/CheckCircleOutlineOutlined'
+import ErrorIcon from '@material-ui/icons/Error'
+import WarningIcon from '@material-ui/icons/Warning'
+import { green, yellow } from '@material-ui/core/colors'
+
+export interface NetworkIndicatorProps {
+  iconClassName?: string
+  currentNetworkId?: number
+  requiredNetworkId?: number
+  onRightNetworkMessage?: string
```
Comment (vojtechsimetka) [inline src/components/atoms/NetworkIndicator.tsx:12]: Can we rename this to `onCorrectNetwork` or `onMatchingNetwork`?

File: src/components/atoms/NetworkIndicator.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  onWrongNetworkMessage?: string
```
Comment (vojtechsimetka) [inline src/components/atoms/NetworkIndicator.tsx:13]: Probably `onNetworkMismatch` would be better naming

File: src/components/atoms/NetworkIndicator.tsx (hunk 2 above)
Comment (farinalvaro) [reply src/components/atoms/NetworkIndicator.tsx:35]: Thank you for the suggestion, will apply this change with the new naming 🙂

File: src/components/atoms/NetworkIndicator.tsx (hunk 5 above)
Comment (farinalvaro) [reply src/components/atoms/NetworkIndicator.tsx:12]: > Do we still need to commit the `dist` folder?

Good point. Only for testing purposes but yes, we can finally get rid off that

**Pull request merged**